### PR TITLE
Grouper-IRODS integration changes

### PR DESCRIPTION
**Pull request by ianmcorvidae**

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
I think this is ready enough for review. There may be more changes I still want to make, but this is the basic bits at least.

File: src/clj_jargon/users.clj [hunk 1]
```diff
@@ -55,3 +63,41 @@
   (cond (not (= (.getUserName irodsAccount) (.getProxyName irodsAccount))) true
         (not (= (.getZone irodsAccount) (.getProxyZone irodsAccount))) true
         :else false))
+
+(defn group-exists?
```
Comment (ianmcorvidae) [inline src/clj_jargon/users.clj:67]: I decided to put these in clj-jargon.users but if someone would prefer I make a new namespace I'm not opposed, this was just easier (and groups are both users and consist of users, and there's user-focused group listing functions here already).

File: src/clj_jargon/users.clj [hunk 2]
```diff
@@ -9,13 +9,21 @@
...
+   })
+
 (defn user
   [{^UserAO user-ao :userAO} username]
   (otel/with-span [s ["user" {:attributes {"irods.user" username}}]]
     (try
       (let [jargon-user (.findByName user-ao username)]
         {:id (.getId jargon-user)
-         :type :user ;; better if we use the UserTypeEnum for rodsuser/groupadmin/rodsadmin/rodsgroup/unknown
+         :type (get user-type-mapping (.getTextValue (.getUserType jargon-user)) :unknown)
```
Comment (ianmcorvidae) [inline src/clj_jargon/users.clj:26]: This change is to enable validating that a user is a group admin or full admin. Nothing in data-info uses the type field that I can find, and nothing I could find anywhere checked any value but `:none` (to indicate the user doesn't exist), so I think this shouldn't cause any issues.

**Review by slr71 — APPROVED** [review]
Looks good to me. :shipit:

**Comment by ianmcorvidae** [comment]
I'll go ahead and merge this, though there might be more changes if needed for the sake of reviews on the other PR, so I won't cut a release until that's set.

**Pull request merged**